Providers/Scripts/2.4x-2.5x/Scripts/nxOMSGenerateInventoryMof.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python
+#============================================================================
+# Copyright (C) Microsoft Corporation, All rights reserved. 
+#============================================================================
+
+import os
+import imp
+import re
+import codecs
+import shutil
+import string
+
+protocol = imp.load_source('protocol', '../protocol.py')
+nxDSCLog = imp.load_source('nxDSCLog', '../nxDSCLog.py')
+
+LG = nxDSCLog.DSCLog
+try:
+    import hashlib
+    md5const = hashlib.md5
+except ImportError:
+    import md5
+    md5const = md5.md5
+
+inventoryMof_path = '/etc/opt/microsoft/omsagent/conf/omsagent.d/'
+
+def init_vars(Instances):
+    new_instances = []
+    if Instances is not None :
+        for instance in Instances:
+            if instance['InstanceName'].value is not None:
+                instance['InstanceName']=instance['InstanceName'].value
+
+            if instance['ClassName'].value is not None:
+                instance['ClassName']=instance['ClassName'].value
+
+            new_properties = []
+            if instance['Properties'] is not None and len(instance['Properties'].value) > 0:
+                for property in instance['Properties'].value:
+                    if property is not None and len(property) > 0:
+                        new_properties.append(property)
+
+            if len(new_properties) > 0:
+                instance['Properties'] = new_properties
+                new_instances.append(instance)
+
+    Instances = new_instances
+    
+def Set_Marshall(FileName, Enable = False, Instances = None, RunIntervalInSeconds = 300, Tag = "default", Format = "tsv", FilterType = "filter_changetracking", Configuration = None):
+    init_vars(Instances)
+    Set(FileName, Enable, Instances, RunIntervalInSeconds, Tag, Format, FilterType, Configuration)
+    return [0]
+
+def Test_Marshall(FileName, Enable = False, Instances = None, RunIntervalInSeconds = 300, Tag = "default", Format = "tsv", FilterType = "filter_changetracking", Configuration = None):
+    init_vars(Instances)
+    return Test(Enable, Instances)
+
+def Get_Marshall(FileName, Enable = False, Instances = None, RunIntervalInSeconds = 300, Tag = "default", Format = "tsv", FilterType = "filter_changetracking", Configuration = None):
+    arg_names = list(locals().keys())
+    init_vars(Instances)
+    
+    CurrentInstances = Instances
```

Comment:
This is wrong. The "Instances" is the input. You will get None in the Get. You need to return the current settings from the MOF.